src/rules/unnecessary_function_arguments.erl

```diff
@@ -3,24 +3,111 @@
 %%      ignored in all function clauses.</p>
 %%      <p>To avoid this warning, remove the unused argument(s).</p>
 %%      <p><b>Note:</b> This rule will not emit a warning if the function
-%%      implements a behaviour callback or a NIF call.</p>
+%%      implements a NIF call.</p>
 -module(unnecessary_function_arguments).
 
 -behaviour(hank_rule).
 
 -export([analyze/2, ignored/2]).
 
+%% To allow erl_syntax:syntaxTree/0 type spec
+-elvis([{elvis_style, atom_naming_convention, #{regex => "^([a-zA-Z][a-z0-9]*_?)*$"}},
+        {elvis_style, invalid_dynamic_call, disable},
```

Comment:
Add a comment stating that we have to call `Module:behaviour_info/1` and that's why we need to disable this rule, please.